examples/simple_arm_motion/README.md

```diff
@@ -0,0 +1,141 @@
+```
+The ROS2 code uses service calls to do the same thing:
+```python
+    # Claim robot
+    node.get_logger().info('Claiming robot')
+    result = robot.command('claim')
+    if not result.success:
+        node.get_logger().error('Unable to claim robot message was ' + result.message)
+        return False
+    node.get_logger().info('Claimed robot')
+
+    # Stand the robot up.
+    node.get_logger().info('Powering robot on')
+    result = robot.command('power_on')
+    if not result.success:
+        node.get_logger().error('Unable to power on robot message was ' + result.message)
+        return False
+    node.get_logger().info('Standing robot up')
+    result = robot.command('stand')
+    if not result.success:
+        node.get_logger().error('Robot did not stand message was ' + result.message)
+        return False
+    node.get_logger().info('Successfully stood up.')
+```
+Note that we also use ROS's text logger instead of the Spot API logger.
+
+Once the robot is initialized and standing, the next change to the code comes when we need to get the robot's current position.  With direct API calls, this uses the robot state:
+```python
+        robot_state = robot_state_client.get_robot_state()
+        odom_T_flat_body = get_a_tform_b(robot_state.kinematic_state.transforms_snapshot,
+                                         ODOM_FRAME_NAME, GRAV_ALIGNED_BODY_FRAME_NAME)
+```
+With ROS2, we use TF:
+```python
+    odom_T_flat_body = tf_listener.lookup_a_tform_b(ODOM_FRAME_NAME, GRAV_ALIGNED_BODY_FRAME_NAME)
+```
+
+The commands are built the same way after this, but we change how we send them.  The Spot API code has a function for waiting until the arm has moved to position:
+```python
+        # Send the request
+        cmd_id = command_client.robot_command(command)
+        robot.logger.info('Moving arm to position 1.')
+
+        # Wait until the arm arrives at the goal.
+        block_until_arm_arrives_with_prints(robot, command_client, cmd_id)
+```
+In ROS2, we convert the created protobuf to a ROS2 action goal and we use the action client `send_goal_and_wait` function to replace the `block_until_arm_arrives_with_prints` function (we do not get the printing, but we could echo the action feedback topic or use the non-blocking `send_goal_async` if we wanted to do that):
+```python
+    # Convert to a ROS message
+    action_goal = RobotCommand.Goal()
+    conv.convert_proto_to_bosdyn_msgs_robot_command(command, action_goal.command)
+    # Send the request and wait until the arm arrives at the goal
+    node.get_logger().info('Moving arm to position 1.')
+    robot_command_client.send_goal_and_wait(action_goal)
+```
+
+The last thing to note in this example is the line
+```python
+    tf_listener.shutdown()
+```
+Owing to a bug in how tf threads are handled, if you want programs involving the TF listener wrapper to end cleanly, you have to make this call before the `tf_listener` is destructed.
```

Comment:
Interesting. Not for this PR, but would it make sense to make a `__del__` function?

Reply:
It doesn't work!  I tried every way I could think of but it still hangs.  I suspect the problem is in some sort of order-of-delete thing...  You can mess with it locally with the `tf_listener_wrapper_test` and see if you can figure it out...